packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -38,6 +38,9 @@ export class EngineReportingExtension<TContext = any>
   private options: EngineReportingOptions<TContext>;
   private addTrace: (args: AddTraceArgs) => Promise<void>;
   private generateClientInfo: GenerateClientInfo<TContext>;
+  private shouldReportQuery: (
+    request: GraphQLRequestContext<TContext>,
```

Comment:
Per a comment I'll make below, I think we should have this receive [`GraphQLRequestContextDidResolveOperation`](https://github.com/apollographql/apollo-server/blob/a926b7eedbb87abab2ec70fb03d71743985cb18d/packages/apollo-server-types/src/index.ts#L188-L193) in order to ensure that `operationName` is available, which happens [after parsing and validation](https://github.com/apollographql/apollo-server/blob/6564081240ee884f3d6c5522ed17f1b480a9451c/packages/apollo-server-core/src/requestPipeline.ts#L324-L327)